### ISSUE TITLE
GDScript Typed Autocompletion

### ISSUE THREAD

**Issue by skgalaxy** [godotengine/godot#21340]
**Godot version:**

3.1 master

**Issue description:**

Some collaborators have proposed a great idea from another I had to make it easier and faster to write code in GDScript, but that, however, could not be implemented. It was about adding a shortcut for typed nodes. You could not implement a shortcut that works in the analysis of the code, but you can do something similar only in the editor: Autocomplete the syntax of the already typed nodes. Then another idea came: Now that we have Typed GDScript, can we improve code autocompletion in more ways?

- Autocompleting the types of the nodes in relation to the current scene
- Autocompleting the resource preload
- Other things that you can think of

And once this is done you have to think about how to implement it. An option in Project Settings that activates Typed Autocompleter? A special syntax? I would prefer the first, since it does not alter the existing ease of use.

Credits: @Zylann @vnen @OvermindDL1

**Comment by skgalaxy** [comment]
#21189

**Comment by vnen** [comment]
> Autocompleting types of virtual functions (arguments and return type)

This is already done and there's an editor setting for it. It also enable types in the signal connection functions that the editor generates.

**Comment by Jummit** [comment]
Is this still an issue?

**Comment by Calinou** [comment]
Indeed, features in this issue have been implemented. If you have other proposals to make, please open an issue on the [Godot Proposals](https://github.com/godotengine/godot-proposals) repository.
